server/translations.go

```diff
@@ -25,3 +29,83 @@ func (s *Server) handleTranslationPR(ctx context.Context, pr *model.PullRequest)
 		return
 	}
 }
+
+func (s *Server) handleModificationOfLanguageFiles(ctx context.Context, pr *model.PullRequest) {
+	repoConfig, isTranslated := isTranslatedRepo(pr, s.Config.TranslationsRepos)
+	if !isTranslated {
+		return
+	}
+
+	if s.Config.TranslationsBot == pr.Username {
+		return
+	}
+
+	prFiles, err := s.getPRFiles(ctx, pr.RepoOwner, pr.RepoName, pr.Number)
+	if err != nil {
+		mlog.Error("Error listing PR files", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.Err(err))
+		return
+	}
+
+	languageDir, languageBaseFile := path.Split(repoConfig.BaseFilePath)
+
+	if !isModifyingLanguageDir(prFiles, languageDir) {
+		return
+	}
+
+	if isOnlyModifyingAllowedFiles(prFiles, languageDir, languageBaseFile) {
+		return
+	}
+
+	mlog.Info("Modification of language file", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("username", pr.Username))
+	msg := fmt.Sprintf("Thank you for your contribution, @%s! For translating, please use [https://translate.mattermost.com](https://translate.mattermost.com) and join our community on [https://community.mattermost.com/core/channels/localization](https://community.mattermost.com/core/channels/localization) ", pr.Username)
+	s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, msg)
+
+	_, _, err = s.GithubClient.Issues.AddLabelsToIssue(ctx, pr.RepoOwner, pr.RepoName, pr.Number, []string{"Lifecycle/1:stale", "Do Not Merge"})
+	if err != nil {
+		msg := fmt.Sprintf("error applying labels %v:\n ```%v```", pr.URL, err.Error())
+		s.logToMattermost(ctx, msg)
```

Comment:
Just curious: what is the general rule that determines when do we log to mattermost, and when we don't?

Reply:
If it is interesting for me to know. In this case, I am interested that these PRs cannot be merged which are created.

Reply:
I see. If that is the sole intention, would it make sense to have Kibana filters set up and monitor log patterns like these? That is the central destinations for all logs and it is easy to set up ad-hoc filters and alerts dynamically without modifying the code.

My point is that monitoring logs should be a separate concern from the code.

Nothing to do right now. Just wanted your opinion on what do you think.